test/functional/Valhalla/src_lw5/org/openj9/test/lworld/ValueTypeSystemArraycopyTests.java

```diff
@@ -59,46 +59,26 @@ public static class SomeIdentityClass2 implements SomeInterface {
 	public value static class SomeValueClass implements SomeInterface {
 		double val1;
 		long val2;
-		SomeValueClass2 val3;
+		SomeValueClass2! val3;
 		int val4;
 
+		public implicit SomeValueClass();
+
 		SomeValueClass(int i) {
 			this.val1 = (double)i;
 			this.val2 = (long)(i*2);
-			this.val3 = new SomeValueClass2(i*3);
+			this.val3 = (SomeValueClass2!)new SomeValueClass2(i*3);
 			this.val4 = i*4;
 		}
 	}
 
-	public primitive static class SomeValueClass2 implements SomeInterface {
+	public value static class SomeValueClass2 implements SomeInterface {
 		long val1;
 		double val2;
 
-		SomeValueClass2(int i) {
-			this.val1 = (long)(i*5);
-			this.val2 = (double)(i*6);
-		}
-	}
-
-	public primitive static class SomePrimitiveValueClass implements SomeInterface {
-		double val1;
-		long val2;
-		SomePrimitiveValueClass2 val3;
-		int val4;
-
-		SomePrimitiveValueClass(int i) {
-			this.val1 = (double)i;
-			this.val2 = (long)(i*2);
-			this.val3 = new SomePrimitiveValueClass2(i*3);
-			this.val4 = i*4;
-		}
-	}
-
-	public primitive static class SomePrimitiveValueClass2 implements SomeInterface {
-		long val1;
-		double val2;
+		public implicit SomeValueClass2();
 
-		SomePrimitiveValueClass2(int i) {
+		SomeValueClass2(int i) {
 			this.val1 = (long)(i*5);
 			this.val2 = (double)(i*6);
 		}
```

Comment:
`SomeValueClass` is meant to test just value object class. `SomePrimitiveValueClass` is meant to test Null Restricted flattenable value object class. `SomePrimitiveValueClass` shouldn't be removed. I think `SomeValueClass2` shouldn't be primitive in the original version. The test could be updated below:

```c++
	public value static class SomeValueClass implements SomeInterface {
		double val1;
		long val2;
		SomeValueClass2 val3;
		int val4;

		SomeValueClass(int i) {
			this.val1 = (double)i;
			this.val2 = (long)(i*2);
			this.val3 = new SomeValueClass2(i*3);
			this.val4 = i*4;
		}
	}

	public value static class SomeValueClass2 implements SomeInterface {
		long val1;
		double val2;

		SomeValueClass2(int i) {
			this.val1 = (long)(i*5);
			this.val2 = (double)(i*6);
		}
	}

	public value static class SomeNullRestrictedValueClass implements SomeInterface {
		double val1;
		long val2;
		SomeNullRestrictedValueClass2! val3;
		int val4;

		public implicit SomeNullRestrictedValueClass();

		SomeNullRestrictedValueClass(int i) {
			this.val1 = (double)i;
			this.val2 = (long)(i*2);
			this.val3 = new (SomeNullRestrictedValueClass2!)(i*3);
			this.val4 = i*4;
		}
	}

	public value static class SomeNullRestrictedValueClass2 implements SomeInterface {
		long val1;
		double val2;

		public implicit SomeNullRestrictedValueClass2();

		SomeNullRestrictedValueClass2(int i) {
			this.val1 = (long)(i*5);
			this.val2 = (double)(i*6);
		}
	}
```

Reply:
Thanks for the feedback. I think SomeValueClass could be used for regular value class and nullrestricted case but its probably easier to read this way.

Reply:
I removed primitive from SomeValueClass2 in the qtypes version of the tests and updated the lw5 tests to use SomeNullRestrictedValueClass and SomeNullRestrictedValueClass2